src/visitor.js

```diff
@@ -85,8 +86,21 @@ const render = (
 export const visitElement = (
   element: AbstractElement,
   queue: Frame[],
-  visitor: Visitor
+  visitor: Visitor,
+  visitOptions?: VisitOptions
 ): AbstractElement[] => {
+  // By default, the supplied `visitor` function is only executed on non-string
+  // React elements. Here we call the visitor on all other React component types
+  // if explicitly requested.
+  if (
+    visitOptions !== undefined &&
+    visitOptions.visitAllComponentTypes &&
+    (typeOf(element) !== REACT_ELEMENT_TYPE || typeof element.type === 'string')
+  ) {
+    const el = ((element: any): UserElement | DOMElement)
+    visitor(el)
```

Comment:
Wasn't sure if I should be suspending here, so I went with the simple route to get your feedback first.